passes/source2il.nim

```diff
@@ -514,6 +530,20 @@ proc callToIL(c; t; n: NodeIndex, bu; stmts): SemType =
     bu.add Node(kind: IntVal)
     result = errorType()
 
+proc commonType(a, b: Expr): SemType =
```

Comment:
I think it'd be better for `commonType` operates on `SemType` and is located in `types`. This makes it more generally applicable, and it's also placed together with the other type queries / operations.

Reply:
Sounds good, it wasn't using `Expr` itself, so it makes sense to drop that knowledge.